xapian-applications/omega/index_file.cc

```diff
@@ -583,6 +671,13 @@ index_mimetype(const string & file, const string & urlterm, const string & url,
 	}
     }
 
+    // if the ctime has changed but the mtime is unchanged, we can just
+    // update the existing Document and avoid having to re-extract text, etc.
+    if (use_ctime && d.get_ctime() != d.get_mtime()) {
```

Comment:
Comparing the current `ctime` and `mtime` seems here odd - shouldn't we be comparing each separately against the previously value we stored for it?  Or am I missing some trick?

Reply:
Actually, at this point I know that I have to update the database. So if the current `mtime` and `ctime` are equal I know that the content of the file has changed (and I can avoid calling `index_update_entry` to check this).